### ISSUE TITLE
6ie Thermostat

### ISSUE THREAD

**Issue by peteretep** [ha-warmup/warmup#33]
Hello,  I have recently had a 6ie mini thermostat installed in my kitchen alongside a 4ie for my bathroom.
The 4ie is working welll with this component.
So far I have been unable to connect the 6ie to my online warmup account. The android app seems to be very buggy!

Hopefully they will fix it soon - has anyone had success connecting a 6ie to their accound and using it with this Home Assistant component?

**Comment by peteretep** [comment]
I am managing to get the data in charles proxy for the first few requests, but when I get to the add room page the phone doesn't seem to want to connect to the wifi network (limited connection). Any ideas?

**Comment by peteretep** [comment]
yes it's something along those lines but I don't seem to be able to get it to reconnect to the wifi network without it complaining. I'm on the add room page and it's using my 3g network. If I manually reconnect to the wifi network the warmuip app complains of a network problem.
So frustrating!
Thanks for your help anyway.

**Comment by tboothman** [comment]
I've not seen an onscreen keyboard on the 6ie. You configure the wifi using the app

**Comment by artmg** [comment]
Thanks @sheppy for confirming your positive testing - enjoy your warm feet this coming winter

**Comment by emtunc** [comment]
> Just an FYI - I have just installed a 6ie and used this addon and it appears in HA fine

Cheers sheppy 😉 I have two 6ie's that I'm about to link up 🤞

**Comment by 3ll3d00d** [comment]
I found another limitation, at least using the android app to configure it. Their support say the 6ie supports a password upto 125 chars long but I found it can't cope with an & in it

**Comment by etsweeney** [comment]
I'm having lots of issues trying to connect my 6ie to my wifi. Using my android phone. SSID is <32 chars and Password is <15 chars. I have google wifi which has dual band, so that should'nt be an issue either. Very frustrating to have a bathroom remodel and this feature is not available. Just posting this here in case anyone else has a solution I can try.

**Comment by artmg** [comment]
Other suggestions people have made include:
* wifi PSK or password less than 15 chars and avoid ` & `
   - maybe try avoiding other odd characters until you get a connection working
* try different room choices (e.g. 'kitchen') to get it connected, and then change afterwards from https://my.warmup.com/home
* using a web proxy application like `charles` to diagnose failures during configuration (quite advanced procedure)
* if you are using the Android app consider trying a friend's iPhone app instead
  - also consult the reviews on both app stores – even if ratings seem to be universally low, they do sometimes have useful suggestions 
     - I've seen some suggestions that you need good mobile internet connectivity as well as wifi!

Unfortunately until you get your device onto the network you cannot configure plugins like ours, but I do wish you luck in finding the magic conjuration that gets your heat going

**Comment by etsweeney** [comment]
@artmg Thanks for the tips! I've tried all of those (except charles) to no avail.

I ended up calling warmup support and they indicated that they've been having issues with the app and plan to have an update available soon (in days, they said). However, they haven't updated that app in a long time, so I found that hard to believe. Will check again in a few days to see if there's any changes.

**Comment by hawtvawt** [comment]
Eventually managed to get two 6iE connected, took several attempts with the app involving switching on and off when you see the 201 error.

**Comment by artmg** [comment]
Thanks for your confirmation and troubleshooting suggestions, @hawtvawt  - I removed the duplication of your issue from here as you clearly found [the correct place to raise it](https://github.com/ha-warmup/warmup/issues/new).

**Comment by tijnhoekstra** [comment]
> Eventually managed to get two 6iE connected, took several attempts with the app involving switching on and off when you see the 201 error.

What do mean by switching on and off, exactly? After the 201 I can only reset the thermostat from the menu to retry. 
I also get the W 201 error. Even when I set one phone as mobile hotspot and use another mobile phone to register the room and thermostat by the app for the 6ie.

**Comment by hawtvawt** [comment]
Remove power from the device. On each occasion let it boot up and then go into settings and turn the network connection back on. It will connect to the network and then wait for upto 5 and it will then authenticate with the cloud service.  I tried the help desk and they are useless. 


 I also tried the hotspot route and that failed with the same error message.

**Comment by tijnhoekstra** [comment]
Thanks! Finally it worked to get around the W 201 error.
The trick:
On the 201, keep the mobile app open and leave as it is.
Reboot the thermostat by killing the power or switch off in the menu.
After the reboot, enable network with the toggle and wait for the connection with the cloud.
Only then return to the app and it will ask you what kind of room you want to register.

**Comment by artmg** [comment]
Well done @tijnhoekstra  :100: and thanks for sharing your technique to help others!

**Comment by etsweeney** [comment]
Hey just following up here - I gave up trying to get tour 6IE connected to wifi and home assistant. But just tried turning it on again and was able to connect to it by using their app. I'm not sure if they fixed something on the app side or not, but it was super easy this time. So if anyone else ran into constant issues, maybe try again?

Thanks again, very happy to have this in home assistant now!

**Comment by artmg** [comment]
That's great news - it looks like your call to support last year might have had some impact, albeit slowly. Glad you got your 6IE working in time for winter (assuming you're in a northern region), and thanks for sharing
